### ISSUE TITLE
Run GOTC on AWS

### ISSUE THREAD

**Issue by ruchim** [broadinstitute/cromwell#1542]

**Comment by mcovarr** [comment]
Status update:

tl;dr If one had to try to run the single sample pipeline on AWS, use a single i3.16xlarge spot instance, SUSE ECS optimized Linux, user data something like [this](https://github.com/broadinstitute/aws-backend-private/blob/master/scripts/suse-monster.sh), and otherwise follow the [lengthy script](https://docs.google.com/document/d/1qwY0QBo04WIAvsACbvtIshfxbs1OJRvNp93oclxCRk8/edit).  But there’s still very little chance (like < 2%) that the workflow will succeed with full size BAMs.


SEGVs

Amazon Linux ECS optimized (currently 64-bit amzn-ami-2016.09.g-amazon-ecs-optimized - ami-275ffe31) produces SEGVs with a consistency that makes it rare for workflows to survive to the MarkDuplicates step.  No known workaround.

Action: Don’t use Amazon Linux


EFS

EFS is currently several times slower for most tasks and orders of magnitude slower for MarkDuplicates.  No known workaround.

Action: Don’t use EFS, use a single-machine cluster.


NVMe drive disconnection

SUSE Linux has a bug where the NVMe drives on i3 instance types (my preferred instance type with lots of fast instance storage) are randomly disconnected at boot time.

Action: Live with it, harden the user data script to roll with whatever drives are available.  Post in the forums.  Nick reported this is a known problem and will be fixed upstream in SUSE eventually, but Amazon doesn’t control this AMI.



Random Docker CannotStartContainerErrors

SUSE Linux consistently gets further in the workflow than Amazon Linux, but also consistently exhibits CannotStartContainerErrors deeper into the workflow.

Action: Post in the forums.  Although SUSE is more stable than Amazon Linux, there’s still not enough stability to run the workflow

**Comment by mcovarr** [comment]
Closing per PO request.

**Comment by dinvlad** [comment]
EFS shines only if it's multi-TB, otherwise directly accessing S3 is much faster (and there's no need to first copy the data from S3 to a filesystem when it can be streamed).
